derp/derp_server.go

```diff
@@ -226,6 +230,24 @@ func (s *Server) PrivateKey() key.Private { return s.privateKey }
 // PublicKey returns the server's public key.
 func (s *Server) PublicKey() key.Public { return s.publicKey }
 
+// emptyBytes returns a (possibly-reused) byte slice of the given length.
+func emptyBytes(length int) []byte {
+	bp, _ := bytesPool.Get().(*[]byte)
```

Comment:
the thing you want to return to the pool is the exact pointer you got out.

if you're returning a []byte slice header and putting a different one back in, you're making little 24 byte allocations

So I'm a little skeptical this improves much.

Reply:
I've changed it to return a pointer to the byte slice, which seems to work better. I've also updated the metrics in the comment/PR so that it reflects it